examples/lighting-app/silabs/src/AppTask.cpp

```diff
@@ -86,6 +86,9 @@ CHIP_ERROR AppTask::Init()
         appError(err);
     }
 
+    // Registers Scene handlers for the On/Off cluster on the server
+    app::Clusters::Scenes::ScenesServer::Instance().RegisterSceneHandler(OnOffServer::Instance().GetSceneHandler());
```

Comment:
Because the app layer should be able to implement its own handlers, say it would like to implement one single handler for all clusters. The idea was the there needs to be a way for the user to decide wether to use the default handlers or not and that should not involve having to modify the cluster code. I could add a config : CHIP_USE_DEFAULT_SCENE_HANDLERS but wouldn't that be less explicit?

Reply:
I think ideally things would "work out of the box", and then the app can customize them as needed.  The need to make this `RegisterSceneHandler` call for Scenes to work at all for On/Off is _very_ non-obvious.

So I think the right thing to do is to have On/Off handle this by default but have a compile-time or run-time option to turn that off for applications that want to control things more directly.